### PR TITLE
[Website] Rust release notes / blog post

### PR DESCRIPTION
**Pull request by andygrove**

### CODE REVIEW

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+# Core Arrow Crate
+
+## Iterator Trait
+
+- Primitive arrays (e.g. array of integers) can now be converted to, and initialized from, an iterator. This exposes a 
+very popular API - iterators - to arrow arrays. Work for other types will continue throughout 3.0.0.
+
+## Improved Variable-sized Arrays
+
+- Variable sized arrays (e.g. array of strings) have been internally refactored to more easily support their larger (64
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:45]:
```suggestion
- Variable sized arrays (e.g., array of strings) have been internally refactored to more easily aupport their larger (64-bit 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+## Iterator Trait
+
+- Primitive arrays (e.g. array of integers) can now be converted to, and initialized from, an iterator. This exposes a 
+very popular API - iterators - to arrow arrays. Work for other types will continue throughout 3.0.0.
+
+## Improved Variable-sized Arrays
+
+- Variable sized arrays (e.g. array of strings) have been internally refactored to more easily support their larger (64 
+bit size offset) version. This allowed us to generalize some of the kernels to both (32 and 64) versions, and also
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:46]:
```suggestion
size offset) version. This allowed us to generalize some of the kernels to both (32 and 64) versions and 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+## Kernels
+
+There have been numerous improvements in the Arrow compute kernels, including:
+
+- New kernels have been added for string operations, including substring, min, max, concat, and length.
+- Aggregate sum is now implemented for SIMD with a 5x improvement over the non-SIMD operation
+- Many kernels have been improved to support dictionary-encoded arrays
+- Some kernels were optimized for arrays without nulls, making them significantly faster in that case.
+- Many kernels were optimized in the number of memory copies that are needed to apply them, and also on their
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:57]:
```suggestion
- Many kernels were optimized in the number of memory copies that are needed to apply them and also on their 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+implementation.
+
+## Other Improvements
+
+The Array trait now has `get_buffer_memory_size` and `get_array_memory_size` methods for determining the amount of 
+memory allocated for the array.
+
+# Parquet
+
+Significant effort is underway to create a Parquet writer for Arrow data. This work has not been released as part of
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:67]:
```suggestion
A significant effort is underway to create a Parquet writer for Arrow data. This work has not been released as part of 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+- sort
+- collect
+- explain
+
+## Performance & Scalability
+
+DataFusion query execution now uses `async`/`await` with the tokio threaded runtime rather than launching dedicated 
+threads, making queries scale much better across available cores.
+
+The hash aggregate physical operator has been largely re-written, resulting in significant performance improvements.
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:103]:
```suggestion
The hash aggregate physical operator has been primarily re-written, resulting in significant performance improvements.
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+- IsNull and IsNotNull
+- Min/Max for strings (lexicographic order)
+- Array of columns
+- Concatenation of strings
+- Aliases of aggregate expressions
+
+Many existing expressions were also significantly optimized (2-3x speedups) by avoiding memory copies and leveraging 
+Arrow format’s invariants.
+
+Unary mathematical functions (such as sqrt) now support both 32 and 64 bit floats, and return the corresponding type,
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:122]:
```suggestion
Unary mathematical functions (such as sqrt) now support both 32 and 64-bit floats and return the corresponding type, 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+Many existing expressions were also significantly optimized (2-3x speedups) by avoiding memory copies and leveraging 
+Arrow format’s invariants.
+
+Unary mathematical functions (such as sqrt) now support both 32 and 64 bit floats, and return the corresponding type, 
+thereby allowing faster operations when higher precision is not needed.
+
+### Improved User-defined Functions (UDFs)
+The API to use and register UDFs has been significantly improved, allowing users to register UDFs and call them both 
+via SQL and the DataFrame API. UDFs now also have the same generality as DataFusion’s own functions, including variadic
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:127]:
```suggestion
via SQL and the DataFrame API. UDFs now also have the same generality as DataFusion’s functions, including variadic 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+thereby allowing faster operations when higher precision is not needed.
+
+### Improved User-defined Functions (UDFs)
+The API to use and register UDFs has been significantly improved, allowing users to register UDFs and call them both 
+via SQL and the DataFrame API. UDFs now also have the same generality as DataFusion’s own functions, including variadic 
+and dynamically typed arguments.
+
+### User-defined Aggregate Functions (UDAFs)
+DataFusion now supports user-defined aggregate functions that can be used to perform operations than span multiple 
+rows, batches and partitions. UDAFs have the same generality of DataFusion’s own functions and support both row updates
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:132]:
```suggestion
rows, batches, and partitions. UDAFs have the same generality as DataFusion’s functions and support both row updates 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+via SQL and the DataFrame API. UDFs now also have the same generality as DataFusion’s own functions, including variadic 
+and dynamically typed arguments.
+
+### User-defined Aggregate Functions (UDAFs)
+DataFusion now supports user-defined aggregate functions that can be used to perform operations than span multiple 
+rows, batches and partitions. UDAFs have the same generality of DataFusion’s own functions and support both row updates 
+and batch updates. You can check out [this example][5] to learn how to declare and use a UDAF.
+
+### User-defined Constants
+DataFusion now supports registering constants (e.g. “@version”), that live for the duration of the execution context
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:136]:
```suggestion
DataFusion now supports registering constants (e.g. “@version”) that live for the duration of the execution context 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+It is now possible to see the query plan for a SQL statement using EXPLAIN syntax.
+
+# Benchmarks
+    
+The benchmark crate now contains a new benchmark based on TPC-H that can execute TPC-H query 1 against CSV, Parquet, 
+and memory data sources. This is useful for running benchmarks against larger data sets.
+
+# Integration Testing / IPC
+
+Arrow IPC is the format for serialization and interprocess communication. It is described in arrow.apache.org, and is
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:165]:
```suggestion
Arrow IPC is the format for serialization and interprocess communication. It is described in arrow.apache.org and is 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+# Roadmap for 3.0.0 and Beyond
+
+Here are some of the initiatives that contributors are currently working on for future releases:
+
+- Support stable Rust
+- Improved DictionaryArray support and performance
+- Implement inner equijoins
+- Support for various platforms like ARMv8
+- Supporting the C Data Interface from Rust, to better support interoperability with other Arrow implementations
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:189]:
```suggestion
- Supporting the C Data Interface from Rust to better support interoperability with other Arrow implementations
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+
+# How to Get Involved
+
+If you are interested in contributing to the Rust subproject in Apache Arrow you can find a list of open issues
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:193]:
```suggestion
If you are interested in contributing to the Rust subproject in Apache Arrow, you can find a list of open issues 
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+- Implement inner equijoins
+- Support for various platforms like ARMv8
+- Supporting the C Data Interface from Rust, to better support interoperability with other Arrow implementations
+
+# How to Get Involved
+
+If you are interested in contributing to the Rust subproject in Apache Arrow you can find a list of open issues 
+suitable for beginners here and the full list here.
+
+Other ways to get involved include trying out Arrow on some of your own data and filing bug reports and helping to
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:196]:
```suggestion
Other ways to get involved include trying out Arrow on some of your data and filing bug reports, and helping to 
```

**Review by vertexclique — APPROVED** [review]
Proofread.

File: _posts/2020-10-19-rust-2.0.0-release.md (hunk 10 above)
Comment (jorgecarleitao) [reply _posts/2020-10-19-rust-2.0.0-release.md:165]: and add link?

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 14]
```diff
@@ -0,0 +1,203 @@
...
+the format used for file and stream I/O between applications wishing to interchange Arrow data.
+
+The Arrow project released IPC version 5 of the Arrow IPC format in version 1.0.0. Before that, a message padding 
+change was made in version 0.15.0 to change the default padding to 8 bytes, while remaining in IPC version 4. Arrow 
+release 0.14.1 and earlier were the last releases to use the legacy 4 byte alignment.
+As part of 2.0.0, the Rust implementation was updated to comply with the changes up to release 0.15.0 of Arrow. 
+Work on supporting IPC version 5 is underway, and is expected to be completed in time for 3.0.0.
+
+As part of the conformance work, Rust is being added to the Arrow integration suite, which tests that supported 
+language implementations (ARROW-3690):
```
Comment (jorgecarleitao) [inline _posts/2020-10-19-rust-2.0.0-release.md:175]: link to issue?

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+- Support stable Rust
+- Improved DictionaryArray support and performance
+- Implement inner equijoins
+- Support for various platforms like ARMv8
+- Supporting the C Data Interface from Rust, to better support interoperability with other Arrow implementations
+
+# How to Get Involved
+
+If you are interested in contributing to the Rust subproject in Apache Arrow you can find a list of open issues 
+suitable for beginners here and the full list here.
```
Comment (jorgecarleitao) [inline _posts/2020-10-19-rust-2.0.0-release.md:194]: here and here missing links?

File: _posts/2020-10-19-rust-2.0.0-release.md (hunk 10 above)
Comment (vertexclique) [reply _posts/2020-10-19-rust-2.0.0-release.md:165]: Oh yeah most probably

File: _posts/2020-10-19-rust-2.0.0-release.md (hunk 14 above)
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:175]:
```suggestion
language implementations [(ARROW-3690)][7]:
```

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 16]
```diff
@@ -0,0 +1,203 @@
...
+
+Other ways to get involved include trying out Arrow on some of your own data and filing bug reports and helping to 
+improve the documentation.
+
+[1]: https://issues.apache.org/jira/browse/ARROW-10295?jql=project%20%3D%20ARROW%20AND%20status%20not%20in%20%20(Open%2C%20%22In%20Progress%22%2C%20Reopened)%20and%20fixVersion%20%3D%202.0.0%20AND%20text%20~%20rust%20ORDER%20BY%20created%20DESC
+[2]: https://github.com/apache/arrow/tree/rust-parquet-arrow-writer
+[3]: https://github.com/apache/arrow/tree/master/rust/parquet_derive
+[4]: https://docs.rs/datafusion/2.0.0/datafusion/dataframe/trait.DataFrame.html
+[5]: https://github.com/apache/arrow/blob/master/rust/datafusion/examples/simple_udaf.rs
```
Comment (vertexclique) [inline _posts/2020-10-19-rust-2.0.0-release.md:203]:
```suggestion
[5]: https://github.com/apache/arrow/blob/master/rust/datafusion/examples/simple_udaf.rs
[6]: https://arrow.apache.org/
[7]: https://issues.apache.org/jira/browse/ARROW-3690
```

**Review by vertexclique — COMMENTED** [review]
Added links

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 17]
```diff
@@ -0,0 +1,203 @@
+---
+layout: post
+title: "Apache Arrow 2.0.0 (Rust)"
```
Comment (nealrichardson) [inline _posts/2020-10-19-rust-2.0.0-release.md:3]: Can you give this a more expressive title? It's going to look odd next to the general release post https://github.com/apache/arrow-site/pull/79/files#diff-4cab3ca69341d657c5c14187a3a6aed43c08417df10df8d07f34945ed801fe68R3

Comment (andygrove) [reply _posts/2020-10-19-rust-2.0.0-release.md:3]: Sure. I'm not feeling particularly creative right now but off the top of my head ...

- Apache Arrow 2.0.0 Rust Highlights
- Rust release notes for 2.0.0
- Highlights of Apache Arrow 2.0.0 Rust Implementation

If there are no opinions posted I will pick something later today.

**Comment by andygrove** [comment]
@nealrichardson I think this is good to go

**Comment by andygrove** [comment]
@jorgecarleitao @vertexclique I merged the suggestions but had to fix a couple of things afterward. Let's get this merged and we can follow up with PRs for any additional changes required.

**Review by jorgecarleitao — APPROVED** [review]
LGTM. Thanks for driving this, @andygrove !

**Comment by vertexclique** [comment]
Sure, let's go for it!

File: _posts/2020-10-19-rust-2.0.0-release.md [hunk 18]
```diff
@@ -0,0 +1,207 @@
...
+
+The Logical Plan enum is now extensible through an Extension variant which accepts a UserDefinedLogicalPlan trait using 
+dynamic dispatch. Consequently, DataFusion now supports user-defined logical nodes, thereby allowing complex nodes to 
+be planned and executed. You can check this example to learn how to declare a new node.
+
+### Predicate push-down
+
+DataFusion now has a Predicate push-down optimizer rule that pushes filter operations as close as possible to scans, 
+thereby speeding up the physical execution of suboptimal queries created via the DataFrame API.
+SQL
```
Comment (nevi-me) [inline _posts/2020-10-19-rust-2.0.0-release.md:151]:
```suggestion

### SQL
```

**Review by nevi-me — APPROVED** [review]

File: _posts/2020-10-19-rust-2.0.0-release.md (hunk 17 above)
Comment (nevi-me) [reply _posts/2020-10-19-rust-2.0.0-release.md]: I like "Apache Arrow 2.0.0 Rust Highlights"

**Comment by andygrove** [comment]
@nealrichardson I don't remember the process for merging PRs in this repo. Do we just hit the merge button in Github?

**Pull request merged**